### PR TITLE
prevent n+1 cost queries

### PR DESCRIPTION
**Pull request by machisuji**
WP [#23782](https://community.openproject.com/work_packages/23782/activity)

### CODE REVIEW

File: lib/open_project/costs/engine.rb [hunk 1]
```diff
@@ -353,6 +353,15 @@ class Engine < ::Rails::Engine
                                                attribute: :updated_on
     end
 
+    module EagerLoadedCosts
+      def eager_loaded_work_packages(ids)
```
Comment (ulferts) [inline lib/open_project/costs/engine.rb:357]: This changes the signature of the original method (parameter vs. no parameter)

**Comment by ulferts** [comment]
Out of lazyness, I didn't want to alter the branches of my plugins and core, I rebased the branch onto dev to see if that would work as well.

Doing that I noticed that parts of the implementation will no longer work when we merge the release brach with dev.  This is due to the changes done to the authorization code in the core and in the costs plugin.

As the change was originally deemed to be developed for the dev branch and to reduce further efforts, I recommend rebasing the PR against the dev branch and fixing the incompatibilities. That would mean that the release/6.0 branch would never see the PRs benefits but that is ok in my book.

**Comment by machisuji** [comment]
Ok will rebase onto dev.

**Comment by machisuji** [comment]
Closed in favour of https://github.com/finnlabs/openproject-costs/pull/261.

**Pull request closed without merging**